### ISSUE TITLE
Create `/contribute` route

### ISSUE THREAD

**Issue by danielbergholz** [danielbergholz/techschool.dev#15]
This is my attempt to automate the process of adding new courses and channels. The idea here is to have a page with an input asking you to add the URL from a YouTube course you like. Elixir will then query the YouTube API and generate all the JSON files for you to simply copy and paste on a PR to add the new course, channel, language, and framework.

Low priority for now. Once the website starts getting more traction, I'll start working on it.

**Comment by danielbergholz** [comment]
Another idea is to automate this process even further. Instead of generating the JSON files for you to copy and open a PR manually, Elixir could do those changes, and then call the GitHub API to open a PR. 

This way the user doesn't even need any programming knowledge (to open a PR) at all, he can just paste the YouTube URL and Elixir will work behind the scenes to open a PR with the course addition.

**Comment by dmitry-sidorov** [comment]
> Another idea is to automate this process even further. Instead of generating the JSON files for you to copy and open a PR manually, Elixir could do those changes, and then call the GitHub API to open a PR.
> 
> This way the user doesn't even need any programming knowledge (to open a PR) at all, he can just paste the YouTube URL and Elixir will work behind the scenes to open a PR with the course addition.

Hey, Daniel! Generating JSON files is not a bad idea at all 😅 

Creating a new PR API request has head and base branches as required params.
https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#create-a-pull-request

That means, we have to create a feature branch first. But I didn't find an API endpoint for it 🥲 
https://docs.github.com/en/rest/branches/branches?apiVersion=2022-11-28#get-a-branch

So, generating JSON for copy-pasting it to PR is the best and only available option for now, from my perspective.

**Comment by danielbergholz** [comment]
Yeah, I also think that creating some JSON files for people to copy and paste will be much easier on our side for now (easier to code)😅
So I think I'll stick with this approach. Thanks for the insights on the GitHub API!
